### PR TITLE
refactor: use market map `Decimal` for queries

### PR DESCRIPTION
**Pull request by aljo242**
Closes: BLO-874

### CODE REVIEW

**Comment by davidterpay** [comment]
needs a rebase

File: x/oracle/keeper/msg_server.go [hunk 1]
```diff
@@ -31,6 +31,10 @@ func (m *msgServer) AddCurrencyPairs(goCtx context.Context, req *types.MsgAddCur
 		return nil, fmt.Errorf("message cannot be empty")
 	}
```
Comment (davidterpay) [inline x/oracle/keeper/msg_server.go:33]: <3

File: x/oracle/keeper/grpc_query.go [hunk 2]
```diff
@@ -78,24 +78,29 @@ func (q queryServer) GetPrice(goCtx context.Context, req *types.GetPriceRequest)
...
 
 	id, ok := q.k.GetIDForCurrencyPair(ctx, cp)
 	if !ok {
-		return nil, fmt.Errorf("no ID found for CurrencyPair: %v", cp)
+		return nil, fmt.Errorf("no ID found for CurrencyPair: %s", cp.String())
+	}
+
+	ticker, err := q.k.mmKeeper.GetTicker(ctx, cp.String())
+	if err != nil {
+		return nil, fmt.Errorf("no ticker for CurrencyPair")
```
Comment (davidterpay) [inline x/oracle/keeper/grpc_query.go]: what happens if the marketmap keeper here is nil?

Comment (davidterpay) [reply x/oracle/keeper/grpc_query.go]: should we just default to the original way of calculating the num decimals?

Comment (aljo242) [reply x/oracle/keeper/grpc_query.go]: yezzir we should

File: x/marketmap/types/codec.go [hunk 3]
```diff
@@ -2,14 +2,27 @@ package types
...
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
 )
 
 // RegisterLegacyAminoCodec registers the necessary x/marketmap interfaces (messages) on the
 // cdc. These types are used for amino serialization.
-func RegisterLegacyAminoCodec(_ *codec.LegacyAmino) {
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	// register the msg-types
+	legacy.RegisterAminoMsg(cdc, &MsgUpdateMarketMap{}, "slinky/x/marketmap/MsgUpdateMarketMap")
+	// legacy.RegisterAminoMsg(cdc, &MsgParams{}, "slinky/x/marketmap/MsgParams")
```
Comment (davidterpay) [inline x/marketmap/types/codec.go]: Is there a reason this is commented out? im assuming the params handler isnt implemented here?

File: x/marketmap/types/codec.go [hunk 4]
```diff
@@ -2,14 +2,27 @@ package types
...
 }
 
 // RegisterInterfaces registers the x/marketmap messages + message service w/ the InterfaceRegistry (registry).
-func RegisterInterfaces(_ types.InterfaceRegistry) {
+func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	// register the implementations of Msg-type
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgUpdateMarketMap{},
+		// &MsgParams{},
```
Comment (davidterpay) [inline x/marketmap/types/codec.go]: same as above.

File: x/marketmap/types/codec.go (hunk 3 above)
Comment (aljo242) [reply x/marketmap/types/codec.go]: ah no this was waiting for the `Params` PR to be merged.  fixing

File: x/marketmap/types/expected_keepers.go [hunk 5]
```diff
@@ -0,0 +1 @@
+package types
```
Comment (davidterpay) [inline x/marketmap/types/expected_keepers.go]: Is this used for anything?

Comment (aljo242) [reply x/marketmap/types/expected_keepers.go]: naw we kill it

File: pkg/types/currency_pair.go [hunk 6]
```diff
@@ -60,7 +60,7 @@ func CurrencyPairFromString(s string) (CurrencyPair, error) {
 
 // Decimals returns the number of decimals that the quote will be reported to. If the quote is Ethereum, then
 // the number of decimals is 18. Otherwise, the decimals will be reorted to 8.
-func (cp *CurrencyPair) Decimals() int {
+func (cp *CurrencyPair) LegacyDecimals() int {
```
Comment (davidterpay) [inline pkg/types/currency_pair.go:63]: bullish

**Review by davidterpay — APPROVED** [review]

**Pull request merged**